### ISSUE TITLE
Bug in multiple layers of Shadow DOM

### ISSUE THREAD

**Issue by slezy** [leongersen/noUiSlider#1121]
Hello there.

I'm trying to use the slider inside of web components based project using [LitElement](https://lit-element.polymer-project.org/).

The slider itself works fine as a separate web component, however if I nest it in another web component, there seems to be a bug on mobile devices where handle doesn't move while sliding, only on click. If I turn off the shadow DOM in wrapper component, slide controls start working again. But then I'll loose all benefits of encapsulation.

You can find the demo here: https://stackblitz.com/edit/lit-element-my-range-yzknl1

**Comment by leongersen** [comment]
Could you try passing a different element to the [`documentElement` option](https://refreshless.com/nouislider/slider-options/#section-document-element)?

**Comment by slezy** [comment]
Sure I can, but I need to select an element within `my-range` component representing the slider. `container` query in the demo is shorthand for something like this `this.shadowRoot.getElementById('range')`.

Point of using web components is encapsulation of markup and styles in one neat reusable bundle that you can use anywhere by importing it's respective `js` file and using custom tag representing the component. Therefor I can't use element from outside of `my-range` component (if that's what you are suggesting), since it doesn't know and care what is outside.

**Comment by tony** [comment]
I can't determine if this is a noUiSlider issue, lit-element, or the implementation itself not being correct. There's no lit-element range sliders at all on GH I could find

@slezy If that is the case, in practice, is it because you are hard-coding the `id=""`? It's an ID, and those are unique per page I think - regardless of it being a web component.  

Does this change anything? https://stackblitz.com/edit/lit-element-my-range-zaqe2v?file=my-range.ts

(Also on the same thing: Wrapping noUiSlider into a webcomponent)

If anyone else with familiarity with lit-element drops by, this would be valuable.

**Comment by slezy** [comment]
I've made another example without lit-element to determine if it is the issue, but behavior of the slider is the same as in lit-element version. https://stackblitz.com/edit/lit-element-my-range-jowt5n?file=my-range.ts
It also seems that the bug is present in desktop chrome mobile view, mobile chrome, samsung internet and safari, but not in the desktop firefox mobile view.

I am actually passing the `id` as a `prop` @tony . There is a default value of course, but consumer of the element can easily override it with his own respective `id`.

Unfortunately it doesn't, in context of the demo it selects the same element as the original query.

**Comment by leongersen** [comment]
It likely only works in one "layer" of shadow DOM due to the fix in PR #1060. I had a quick look around, but I don't see an easy way of accessing the shadowRoot recursively. (quick disclaimer, I haven't worked much with shadow DOM's yet).

**Comment by slezy** [comment]
I finally got time to take a proper look at the issue. You were indeed right @leongersen that it was caused by the way `shadowRoot` was accessed there. This way it took lowermost `shadowRoot` and if there were more of them it naturally did not contain `eventTarget` (since that is contained in the uppermost shadowRoot).
The way of accessing it correctly is through [composedPath](https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath)
